onnx/backend/test/runner/__init__.py

```diff
@@ -316,18 +369,66 @@ def run(test_self: Any, device: str) -> None:
                 and not self.backend.is_compatible(model)
             ):
                 raise unittest.SkipTest("Not compatible with backend")
+
             prepared_model = self.backend.prepare(model, device)
             assert prepared_model is not None
 
-            # TODO after converting all npz files to protobuf, we can delete this.
-            for test_data_npz in glob.glob(os.path.join(model_dir, "test_data_*.npz")):
-                test_data = np.load(test_data_npz, encoding="bytes")
-                inputs = list(test_data["inputs"])
+            if use_dummy:
+                # When the backend test goes through a test involving a
+                # model stored in onnx/backend/test/data/light,
+                # this function generates expected output coming from
+                # from ReferenceEvaluator run with random inputs.
+                # A couple of models include many Conv operators and the
+                # python implementation is slow (such as test_bvlc_alexnet).
+                with open(model_pb_path, "rb") as f:
+                    onx = onnx.load(f)
+                test_data_set = os.path.join(model_dir, "test_data_set_0")
+                if not os.path.exists(test_data_set):
+                    os.mkdir(test_data_set)
+                ref = onnx.reference.ReferenceEvaluator(onx)
+                feeds = {}
+                inits = set(i.name for i in onx.graph.initializer)
+                n_input = 0
+                inputs = []
+                for i in range(len(onx.graph.input)):
+                    if onx.graph.input[i].name in inits:
+                        continue
+                    name = os.path.join(test_data_set, f"input_{n_input}.pb")
+                    inputs.append(name)
```

Comment:
Small fix to `input` here:https://github.com/onnx/onnx/pull/5026. Let me know if it looks good.